[webview_flutter_wkwebview] Add a listener for the canGoBack property change on the iOS platform.

StanleyCocos opened the pull request:
Added a listener callback for the canGoBack property on iOS. Since it can only be effectively implemented on iOS, it is only exposed in the WKWebView layer.

Fixes https://github.com/flutter/flutter/issues/158160

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

cbracken commented:
Hi @StanleyCocos thanks for your patch. I looks like this change is causing existing tests to fail and does not include any new test code. Once you've fixed both those issues, please let me know and I'll take a look!

StanleyCocos commented:
@cbracken 
Thank you for your suggestion. 

I’ve added a test case and fixed the integration test error. However, there seems to be another issue caused by not updating the version. I'm not sure what I should do to fix this. Could you give me some advice?

StanleyCocos commented:
can you review again
cc @cbracken

StanleyCocos commented:
can you review again

cc @cbracken

cbracken commented:
One last fix to make to get CI passing:
```
Changed packages: webview_flutter_wkwebview
[0:00] Running for packages/webview_flutter/webview_flutter_wkwebview...
  No version change.
No version change found, but the change to this package could not be verified to be exempt
from version changes according to repository policy.
If this is a false positive, please comment in the PR to explain why the PR
is exempt, and add (or ask your reviewer to add) the "override: no versioning needed" label.


The following packages had errors:
  packages/webview_flutter/webview_flutter_wkwebview:
    Missing version change
See above for full details.
```

StanleyCocos commented:
I seem to understand that the version information needs to be adjusted. But I don't know how to make the changes. Or should I apply for the "no version control required" tag?

StanleyCocos commented:
I have already added the changelog file content, but the issue seems to still exist. 
cc @cbracken

StanleyCocos commented:
can you review again 
cc @cbracken

cbracken reviewed (approved):
This lgtm. Adding @stuartmorgan as second reviewer.

bparrishMines reviewed (approved):
LGTM

I slightly updated the changelog message and changed it to a minor version bump since this is a new feature.
